### PR TITLE
added SWAPI housekeeping processing and CDF attrs

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

## Overview
Added SWAPI housekeeping processing and CDF attribute
## New Files
- imap_processing/common_cdf_attrs.py
   - To store common attrs

## Updated Files
- imap_processing/swapi/l1/swapi_l1.py
   - added housekeeping processing
- imap_processing/swapi/swapi_cdf_attrs.py
   - Added housekeeping attrs
- imap_processing/swapi/swapi_utils.py
   - removed old create_dataset function and moved it to utils function below and updated it.
- imap_processing/utils.py
## Testing
- imap_processing/tests/swapi/test_swapi_l1.py

### CODE REVIEW

File: imap_processing/utils.py [hunk 1]
```diff
@@ -94,3 +99,64 @@ def convert_raw_to_eu(dataset: xr.Dataset, conversion_table_path, packet_name):
...
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    metadata_arrays = collections.defaultdict(list)
+    description_dict = {}
+
+    for data_packet in packets:
+        # Add metadata to array
+        for key, value in (data_packet.header | data_packet.data).items():
```
Comment (sdhoyt) [inline imap_processing/utils.py:126]: Is this going to cause the CCSDS header to be put into a data array?

File: imap_processing/common_cdf_attrs.py
Comment (bourque) [inline imap_processing/common_cdf_attrs.py:1]: Good idea!

**Review by bourque — APPROVED** [review]
LGTM!

File: imap_processing/utils.py (hunk 1 above)
Comment (tech3371) [reply imap_processing/utils.py:126]: Yes, I have included header in the CDF so far.

Comment (sdhoyt) [reply imap_processing/utils.py:126]: Is that something SWAPI asked for or is that a standard thing we should do? I wasn't planning to put the headers into the CDFs

**Review by sdhoyt — APPROVED** [review]

File: imap_processing/utils.py (hunk 1 above)
Comment (tech3371) [reply imap_processing/utils.py:126]: They didn't ask for this. It's something I have to been carrying information over. But I don't think they will complain if I remove it. If you are thinking of using it and if it helps to remove header, then I am happy to.

**Review by bryan-harter — APPROVED** [review]

**Pull request merged**